src/snowflake/snowpark/_internal/utils.py

```diff
@@ -803,6 +811,22 @@ def get_copy_into_table_options(
     return file_format_type_options, copy_options
 
 
+def get_copy_into_location_options(
+    options: Dict[str, Any]
+) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    """Method that extracts options for COPY INTO LOCATION command into file
+    format type options and copy into location options.
+    """
+    file_format_type_options = options.get("FORMAT_TYPE_OPTIONS", {})
+    copy_into_location_options = options.get("COPY_INTO_LOCATION_OPTIONS", {})
+    for k, v in options.items():
+        if k in COPY_INTO_LOCATION_OPTIONS:
+            copy_into_location_options[k] = v
+        elif k not in NON_FORMAT_TYPE_OPTIONS:
+            file_format_type_options[k] = v
```

Comment:
just for my own learning purposes as I see three categories here:
1. copy into option
2. file format which is not copy into option
3. non file format which is not copy into option

so the options in `copy_into_location_options` doesn't overlap `file_format_type_options` at all?

Reply:
copy into location: https://docs.snowflake.com/en/sql-reference/sql/copy-into-location is used for dataframe writer and,
copy into table: https://docs.snowflake.com/en/sql-reference/sql/copy-into-location is used for dataframe reader.

For the two, COPY OPTIONS supported are slightly different. I renamed the dictionaries so that this distinction is more obvious. let me know if this is better.

Reply:
I see, thanks! it's clear to me now